From cbe9f2c8843a63e5528c188d5251146d4ecc0b22 Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Tue, 15 Oct 2024 14:35:40 +0200
Subject: [PATCH] Fix override of BaseISCSISR.detach

The current detach method of BaseISCSISR doesn't correctly override the method of the SR class.
It actually adds a "delete" parameter, which is seen as an error by analyzers like mypy.

The "delete" parameter was added by this commit:
"iscsi: Delete LUN on detach of RawISCSI"

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/BaseISCSI.py  | 5 ++++-
 drivers/RawISCSISR.py | 2 +-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/BaseISCSI.py b/drivers/BaseISCSI.py
index 71e43ddc..a52442c5 100755
--- a/drivers/BaseISCSI.py
+++ b/drivers/BaseISCSI.py
@@ -433,7 +433,10 @@ class BaseISCSISR(SR.SR):
                 realdev = os.path.realpath(os.path.join(dev_path, dev))
                 util.set_scheduler(os.path.basename(realdev))
 
-    def detach(self, sr_uuid, delete=False):
+    def detach(self, sr_uuid):
+        self.detach_and_delete(sr_uuid, delete=False)
+
+    def detach_and_delete(self, sr_uuid, delete=True):
         keys = []
         pbdref = None
         try:
diff --git a/drivers/RawISCSISR.py b/drivers/RawISCSISR.py
index 0b17cfa9..1df1c7a2 100644
--- a/drivers/RawISCSISR.py
+++ b/drivers/RawISCSISR.py
@@ -64,7 +64,7 @@ class RawISCSISR(BaseISCSI.BaseISCSISR):
         self.managed = True
 
     def detach(self, sr_uuid):
-        super(RawISCSISR, self).detach(sr_uuid, True)
+        super(RawISCSISR, self).detach_and_delete(sr_uuid)
 
     def vdi(self, uuid):
         return ISCSIVDI(self, uuid)
